skorch/callbacks/logging.py

```diff
@@ -728,3 +728,105 @@ def on_epoch_end(self, net, **kwargs):
     def on_train_end(self, net, **kwargs):
         if self.close_after_train:
             self.writer.close()
+
+
+class SacredLogger(Callback):
+    """Logs results from history to Sacred.
+
+    Sacred is a tool to help you configure, organize, log and reproduce
+    experiments developed at IDSIA. See https://github.com/IDSIA/sacred.
+
+    Use this callback to automatically log all interesting values from
+    your net's history to Sacred.
+
+    If you want to log additional information, you can simply add it to
+    ``History``. See the documentation on ``Callbacks``, and ``Scoring`` for
+    more information. Alternatively you can subclass this callback and extend
+    the ``on_*`` methods.
+
+    To use this logger, you first have to install Sacred:
+
+    >>> pip install sacred
+
+    You might also install pymongo to use a mongodb backend. See the upstream_
+    documentation for more details. Once you have installed it, you can set up
+    a simple experiment and pass this Logger as a callback to your skorch
+    estimator:
+
+    >>> from sacred import Experiment
+    >>> from sokrch.callbacks.logging import SacredLogger
+    >>> from sokrch.callbacks.scoring import EpochScoring
+    >>> ex = Experiment()
+
+    >>> @ex.config
+    >>> def my_config():
+    ...     max_epochs = 20
+    ...     lr = 0.01
+
+    >>> @ex.automain
+    >>> def main(_run, max_epochs, lr):
+    ...     # Take care to add additional scoring callbacks *before* the logger.
+    ...     net = NeuralNetClassifier(
+    ...         ClassifierModule,
+    ...         max_epochs=max_epochs,
+    ...         lr=0.01,
+    ...         callbacks=[EpochScoring("f1"), SacredLogger(_run)]
+    ...     )
+    ...     # now fit your estimator to your data
+    ...     net.fit(X, y)
```

Comment:
```suggestion
    # contents of sacred-experiment.py
    >>> import numpy as np
    >>> from sacred import Experiment
    >>> from sklearn.datasets import make_classification
    >>> from skorch.callbacks.logging import SacredLogger
    >>> from skorch.callbacks.scoring import EpochScoring
    >>> from skorch import NeuralNetClassifier
    >>> from skorch.toy import make_classifier

    >>> ex = Experiment()

    >>> @ex.config
    >>> def my_config():
    ...     max_epochs = 20
    ...     lr = 0.01

    >>> X, y = make_classification()
    >>> X, y = X.astype(np.float32), y.astype(np.int64)

    >>> @ex.automain
    >>> def main(_run, max_epochs, lr):
    ...     # Take care to add additional scoring callbacks *before* the logger.
    ...     net = NeuralNetClassifier(
    ...         make_classifier(),
    ...         max_epochs=max_epochs,
    ...         lr=0.01,
    ...         callbacks=[EpochScoring("f1"), SacredLogger(_run)]
    ...     )
    ...     # now fit your estimator to your data
    ...     net.fit(X, y)
    
    Then call this from the command line, e.g. like this:
    ``python sacred-script.py with max_epochs=15``
```

I amended the example so that it can run without further additions. Could you maybe add a sentence to summarize what Sacred does for the user?

Reply:
Is "Sacred is a tool to help you configure, organize, log and reproduce experiments" too broad?